src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -52,11 +66,13 @@ public KafkaMonitor(Map<String, Map> testProps) throws Exception {
         _services.put(name, service);
       }
     }
-
-    _executor = Executors.newSingleThreadScheduledExecutor();
   }
 
-  public void start() {
+  public synchronized void start() {
+    if (_isShutdown) {
```

Comment:
Since you mention this.  Start should only be called once regardless of the first call to start being successful.  Stop and await shutdown should be allowed to be called multiple times, just incase start or shutdown did not complete successfully the first time.  ProduceService and ConsumeService already have something to prevent their start logic from being executed multiple times.